flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -638,6 +638,14 @@ public ProgramTargetDescriptor executeUpdate(String sessionId, String statement)
 			result.close();
 			// catch everything such that the query does not crash the executor
 			throw new SqlExecutionException("Invalid SQL query.", t);
+		} finally {
+			// Remove the temporal table object.
+			context.wrapClassLoader(() -> {
+				final TableEnvironment tableEnvironment = context.getTableEnvironment();
+				tableEnvironment
+						.dropTemporaryTable(String.format("`%s`", jobName.replaceAll("`", "")));
```

Comment:
why remove all "`"？

Reply:
Conflict with the appended one.

Reply:
so the `ObjectIdentifier` in `registerTableSink` is different from `dropTemporaryTable`, even they're generated from the same `String`?

Reply:
Yes, you are right, for `dropTemporaryTable` we actually use the sql-parser to parse the identifier, so it should be valid or legal.